### ISSUE TITLE
Several `Operator` methods have been modified in PennyLane master

### ISSUE THREAD

**Issue by josh146** [aws/amazon-braket-pennylane-plugin-python#87]
**Describe the bug**

In the recent PennyLane `master` commit, several of our operator methods have been modified.

*When creating operators:*

- Matrices: Operators should now define the static method `compute_matrix()` rather than the class method `_matrix()`

- Decompositions: Operators should now define the static method `compute_decompositions()` rather than the static method `decompositions()`.

This will require a small re-naming in https://github.com/aws/amazon-braket-pennylane-plugin-python/blob/main/src/braket/pennylane_plugin/ops.py.

*When accessing operator information:*

- `Operator.matrix` will raise a deprecation warning - use `qml.matrix(op)` instead
- `Operator.eigvals` will raise a deprecation warning - use `qml.eigvals(op)` instead
- `Operator.terms` is now a _method_, `Operator.terms()`.

This will require updates in the following places:

* https://github.com/aws/amazon-braket-pennylane-plugin-python/blob/fd5f5f04292840b8fc5e9abfa73c521bcd4bb547/src/braket/pennylane_plugin/translation.py#L392

* [test/unit_tests/test_ops.py‎](https://cs.github.com/aws/amazon-braket-pennylane-plugin-python/blob/355aafc2b789068399e9395ea21716ecb79e6b1f/test/unit_tests/test_ops.py?q=._matrix)

* [test/integ_tests/test_apply.py‎](https://cs.github.com/aws/amazon-braket-pennylane-plugin-python/blob/355aafc2b789068399e9395ea21716ecb79e6b1f/test/integ_tests/test_apply.py?q=._matrix)

* [src/braket/pennylane_plugin/translation.py‎](https://cs.github.com/aws/amazon-braket-pennylane-plugin-python/blob/355aafc2b789068399e9395ea21716ecb79e6b1f/src/braket/pennylane_plugin/translation.py?q=terms)

**Comment by kshitijc** [comment]
Thanks for the heads up @josh146! A couple of questions:
* When is the Pennylane release with these changes scheduled to go out? 
* Are there any tests that Pennylane recommends the plugin integrators to execute, which would help ensure that they have migrated all functionality properly?

**Comment by josh146** [comment]
No worries @kshitijc!

> When is the Pennylane release with these changes scheduled to go out?

The current release is planned for March 15, and we're happy to help out in any way.

> Are there any tests that Pennylane recommends the plugin integrators to execute, which would help ensure that they have migrated all functionality properly?

Yes! If you install the `master` branch of PennyLane, there are device integration tests available under the CLI utility `pl-device-tests`.

Assuming that the `main` branch of PL-Braket is installed, the development version of PL-Braket can be tested against the development version of PennyLane like so:

```bash
$ pl-device-test --device=braket.local.qubit --tb=short --skip-ops --shots=20000
```

where `--skip-ops` just instructs the integration test to skip any tests using operations unsupported by the device.

At the moment, the device integration tests are showing the current failures:

```
FAILED pennylane/devices/tests/test_measurements.py::TestSupportedObservables::test_supported_observables_can_be_implemented[device_kwargs0-Hamiltonian]
FAILED pennylane/devices/tests/test_measurements.py::TestHamiltonianSupport::test_hamiltonian_diff[device_kwargs0]
```

which directly relate to the fix needed in [src/braket/pennylane_plugin/translation.py‎](https://cs.github.com/aws/amazon-braket-pennylane-plugin-python/blob/355aafc2b789068399e9395ea21716ecb79e6b1f/src/braket/pennylane_plugin/translation.py?q=terms).

This is the only fix needed in the main codebase; the other changes simply relate to the plugin unit tests.

**Comment by josh146** [comment]
No problem @kshitijc! We run these device integration tests daily here: https://github.com/PennyLaneAI/plugin-test-matrix

**Comment by josh146** [comment]
Some other changes that have since been made in `master`:

- `pennylane.measure` has been renamed to `pennylane.measurements` due to an unfortunate naming clash.

- `pennylane.ops.qubit.four_term_grad_recipe` has been removed, as we no longer rely on hardcoded gradient recipes, and instead use the general shift rule. The four term grad recipe can thus be computed programmatically:

  ```python
  >>> op = qml.CRX(0.5, wires=[0, 1])
  >>> coeffs, shifts = qml.gradients.generate_shift_rule(op.parameter_frequencies[0])
  ```

**Comment by christianbmadsen** [comment]
@virajvchaudhari can this issue be closed now that https://github.com/aws/amazon-braket-pennylane-plugin-python/pull/91 has been merged?

**Comment by virajvchaudhari** [comment]
We have addressed the required changes for PL-0.22 with complete tox coverage for PL-Braket plugin in this PR https://github.com/aws/amazon-braket-pennylane-plugin-python/pull/91.
